services/frontend-service/src/ui/components/LockDisplay/LockDisplay.test.tsx

```diff
@@ -117,3 +122,76 @@ describe('Test Auxiliary Functions for Lock Display', () => {
         });
     });
 });
+
+describe('Test delete lock button', () => {
+    interface dataT {
+        name: string;
+        actions: BatchAction[];
+        expectedNumOfActions: number;
+        date: Date;
+    }
+    const lock = {
+        date: new Date('2/1/22'),
+        environment: 'test-env',
+        // application?: string,
+        message: 'string',
+        lockId: 'test-lock-id',
+        authorName: 'test-auth',
+        authorEmail: 'test-mail',
+    };
+    const data: dataT[] = [
+        {
+            name: 'Test environment delete button',
+            date: new Date('2/1/22'),
+            actions: [
+                {
+                    action: {
+                        $case: 'createEnvironmentLock',
+                        createEnvironmentLock: {
+                            environment: 'nmww',
+                            lockId: 'test-lock-id',
+                            message: 'test-message',
+                        },
+                    },
+                },
+                {
+                    action: {
+                        $case: 'createEnvironmentApplicationLock',
+                        createEnvironmentApplicationLock: {
+                            environment: 'nmww',
+                            application: 'test-application',
+                            lockId: 'test-lock-id',
+                            message: 'test-message',
+                        },
+                    },
+                },
+            ],
+            expectedNumOfActions: 1,
+        },
+    ];
+
+    describe.each(data)('', (testcase) => {
+        it('Cart initially empty', () => {
+            render(<LockDisplay lock={lock} />);
+            expect(document.getElementsByClassName('mdc-drawer-sidebar-list').length).toBe(0);
+        });
+        it(testcase.name, () => {
+            const api = useApi;
+            // given
+            updateActions(testcase.actions);
+            api.batchService().ProcessBatch(testcase.actions);
+            // when
+            // const { container } = getWrapper({});
+            render(<LockDisplay lock={lock} />);
+            const result = document.querySelector('.service-action--delete')! as HTMLElement;
+            act(() => {
+                result.click();
+            });
+            render(<SideBar />);
```

Comment:
please remove the SideBar from the Lock display test. 
unit tests are small and only test specific functions. if the sidebar changes, this test should not get affected. 

use mocks to mock any inputs and outputs of this component. 
(i.e. mock the addiction function and expect it to be called with the delete lock action)